include/seqan3/search/algorithm/detail/search_trivial.hpp

```diff
@@ -75,7 +72,7 @@ namespace seqan3::detail
  *
  * ### Exceptions
  *
- * No-throw guarantee if this is also guaranteed when invoking the delegate.
+ * No-throw guarantee if invoking the delegate also guarantees no-throw.
```

Comment:
but then I propose to add a noexcept expression that checks if the call to the delegate is also noexcept.

Reply:
I added `noexcept(noexcept(delegate))` for the two cases